lib/ansible/modules/windows/win_secedit.ps1

```diff
@@ -0,0 +1,199 @@
+                $value = [int]$value
+            } elseif ($value.StartsWith('"') -and $value.EndsWith('"')) {
+                $value = $value.Substring(1, $value.Length - 2)
+            }
+
+            $ini.$section.$name = $value
+        }
+    }
+
+    $ini
+}
+
+$secedit_ini_path = [IO.Path]::GetTempFileName()
+# while this will technically make a change to the system in check mode by
+# creating a new file, we need these values to be able to do anything
+# substantial in check mode
+$export_result = Run-SecEdit -arguments @("/export", "/cfg", $secedit_ini_path, "/quiet")
+
+# check the return code and if the file has been populated, otherwise error out
+if (($export_result.rc -ne 0) -or ((Get-Item -Path $secedit_ini_path).Length -eq 0)) {
+    Remove-Item -Path $secedit_ini_path -Force
+    Fail-Json $result "Failed to export secedit.ini file to $($secedit_ini_path).`nRC: $($export_result.rc)`nSTDOUT: $($export_result.stdout)`nSTDERR: $($export_result.stderr)`nLOG: $($export_result.log)"
+}
+$secedit_ini = ConvertFrom-Ini -file_path $secedit_ini_path
+Remove-Item -Path $secedit_ini_path -Force
+
+if (-not ($secedit_ini.ContainsKey($section))) {
+    Fail-Json $result "The section '$section' does not exist in SecEdit.exe output ini"
+}
+
+$will_change = $false
+if ($secedit_ini.$section.ContainsKey($key)) {
+    $current_value = $secedit_ini.$section.$key
+    $result.before_value = $current_value
+
+    if ($current_value -cne $value) {
+        if ($diff_mode) {
+            $result.diff.prepared = @"
+[$section]
+-$key = $current_value
++$key = $value
+"@
+        }
+
+        $secedit_ini.$section.$key = $value
+        $will_change = $true
+    }
+} else {
+    if ($diff_mode) {
+        $result.diff.prepared = @"
+[$section]
++$key = $value        
+"@
+    }
+    $result.before_value = $null
+    $secedit_ini.$section.$key = $value
+    $will_change = $true
+}
+
+if ($will_change) {
+    $ini_contents = ConvertTo-Ini -ini $secedit_ini
+    Set-Content -Path $secedit_ini_path -Value $ini_contents -WhatIf:$check_mode
+    $result.changed = $true
+
+    if (-not $check_mode) {
+        $secedit_db_path = [IO.Path]::GetTempFileName()
+        Remove-Item -Path $secedit_db_path -Force # needs to be deleted for SecEdit.exe /import to work
+
+        $import_result = Run-SecEdit -arguments @("/configure", "/db", $secedit_db_path, "/cfg", $secedit_ini_path, "/quiet")
+        $result.import_log = $import_result.log
+        Remove-Item -Path $secedit_ini_path -Force
+        if ($import_result.rc -ne 0) {
+            Fail-Json $result "Failed to import secedit.ini file from $($secedit_ini_path).`nRC: $($import_result.rc)`nSTDOUT: $($import_result.stdout)`nSTDERR: $($import_result.stderr)`nLOG: $($import_result.log)"
+        }
+
+        # secedit doesn't error out on improper entries, re-export and verify
+        # the changes occurred
+        $export_result = Run-SecEdit -arguments @("/export", "/cfg", $secedit_ini_path, "/quiet")
```

Comment:
same as win_user_right: couldn't we just factor out the change detection, re-run, and fail on change?